migrations/20210927094004_payment/migration.sql

```diff
@@ -0,0 +1,283 @@
+
+-- CreateEnum
+CREATE TYPE "vault_type" AS ENUM ('personal', 'campaign');
+
+-- CreateEnum
+CREATE TYPE "document_type" AS ENUM ('invoice', 'receipt', 'medical_record', 'other');
+
+-- CreateEnum
+CREATE TYPE "transaction_status" AS ENUM ('initial', 'invalid', 'incomplete', 'declined', 'waiting', 'cancelled', 'succeeded', 'deleted', 'refund', 'paymentRequested');
+
+-- CreateEnum
+CREATE TYPE "payout_type" AS ENUM ('cash', 'bank');
+
+-- CreateEnum
+CREATE TYPE "payout_status" AS ENUM ('initial', 'invalid', 'incomplete', 'declined', 'cancelled', 'succeeded');
+
+-- CreateEnum
+CREATE TYPE "account_holder_type" AS ENUM ('individual', 'company');
+
+-- CreateEnum
+CREATE TYPE "bank_account_status" AS ENUM ('new', 'validated', 'verified', 'verification_failed', 'errored');
+
+-- AlterEnum
+BEGIN;
+CREATE TYPE "beneficiary_type_new" AS ENUM ('individual', 'company');
+ALTER TABLE "beneficiaries" ALTER COLUMN "type" TYPE "beneficiary_type_new" USING ("type"::text::"beneficiary_type_new");
+ALTER TYPE "beneficiary_type" RENAME TO "beneficiary_type_old";
+ALTER TYPE "beneficiary_type_new" RENAME TO "beneficiary_type";
+DROP TYPE "beneficiary_type_old";
+COMMIT;
+
+-- AlterEnum
+-- This migration adds more than one value to an enum.
+-- With PostgreSQL versions 11 and earlier, this is not possible
+-- in a single migration. This can be worked around by creating
+-- multiple migrations, each migration adding only one value to
+-- the enum.
+
+
+ALTER TYPE "person_relation" ADD VALUE 'myself';
+ALTER TYPE "person_relation" ADD VALUE 'myorg';
+
+-- DropForeignKey
+ALTER TABLE "beneficiaries" DROP CONSTRAINT "beneficiaries_person_id_fkey";
+
+-- AlterTable
+ALTER TABLE "beneficiaries" ADD COLUMN     "company_id" UUID,
+ALTER COLUMN "person_id" DROP NOT NULL;
+
+-- AlterTable
+ALTER TABLE "campaigns" ADD COLUMN     "vault_id" UUID,
+DROP COLUMN "currency",
+ADD COLUMN     "currency" "currency" DEFAULT E'BGN';
+
+-- AlterTable
+ALTER TABLE "coordinators" ADD COLUMN     "created_at" TIMESTAMPTZ(6) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+ADD COLUMN     "updated_at" TIMESTAMPTZ(6);
+
+-- AlterTable
+ALTER TABLE "people" ADD COLUMN     "personal_number" TEXT,
+ADD COLUMN     "vault_id" UUID,
+ALTER COLUMN "address" SET DATA TYPE VARCHAR(100);
+
+-- DropEnum
+DROP TYPE "Currency";
+
+-- CreateTable
+CREATE TABLE "companies" (
+    "id" UUID NOT NULL DEFAULT gen_random_uuid(),
+    "company_name" VARCHAR(100) NOT NULL,
+    "companyNumber" TEXT NOT NULL,
+    "legal_person_name" TEXT,
+    "country_code" CITEXT,
+    "city_id" UUID,
+    "created_at" TIMESTAMPTZ(6) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updated_at" TIMESTAMPTZ(6),
+
+    CONSTRAINT "companies_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "benefactors" (
+    "id" UUID NOT NULL DEFAULT gen_random_uuid(),
+    "person_id" UUID NOT NULL,
+    "customer_id" TEXT,
```

Comment:
is the customerID something that the payment provider will give us to keep as reference, or the payment provider needs we to provide an id that we identify the user with? If the latter, then this is the id uuid field and we don't need of separate customerID field in this table

Reply:
@igoychev Yep, the customer id comes from the 3rd party payment provider and is plain string hash like `"id": "cus_HTcX9Re3TfeC0y"`

I assume all payment providers have such customer reference

https://stripe.com/docs/api/customers/object#customer_object-id


So far we start with only one customer id, but those can be extracted in separate relation when we see the need for it.

I didn't spent too much time on this part, so it's open for extension and improvements

Reply:
got it, let's leave the customer_id, just please make it varchar